### PR TITLE
Release1.2.4

### PR DESCRIPTION
**Pull request by nfwvogt**
Update to pyo3 0.18

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 This changelog track changes to the qoqo project starting at version v0.5.0
 
+## 1.2.4
+
+* Updated to pyo3 0.18, qoqo_calculator 1.2.0 and nalgebra 0.32
+
```
Comment (kbarkhqs) [inline CHANGELOG.md:8]: I think it would be good to mention moving to the centralised github workflows

Comment (kbarkhqs) [reply CHANGELOG.md:8]: Also, this should say `qoqo_calculator 1.1.1` I believe

File: qoqo/tests/integration/circuit.rs [hunk 2]
```diff
@@ -43,7 +43,7 @@ fn populate_circuit_rotatex(
     let rotatex_type = py.get_type::<RotateXWrapper>();
     for i in start..stop {
         let new_rotatex_0 = rotatex_type.call1((i, i)).unwrap();
-        // .cast_as::<PyCell<RotateXWrapper>>()
+        // .downcast::<PyCell<RotateXWrapper>>()
```
Comment (kbarkhqs) [inline qoqo/tests/integration/circuit.rs:46]: These lines can probably be deleted

File: qoqo/tests/integration/devices.rs [hunk 3]
```diff
@@ -39,7 +39,7 @@ fn new_genericdevice() -> Py<PyAny> {
         let arguments = (number_qubits,);
         let device_type = py.get_type::<GenericDeviceWrapper>();
         device_type.call1(arguments).unwrap().into()
-        // .cast_as::<PyCell<PyAny>>()
+        // .downcast::<PyCell<PyAny>>()
         // .unwrap()
```
Comment (kbarkhqs) [inline qoqo/tests/integration/devices.rs:43]: Can be deleted

Comment (kbarkhqs) [reply qoqo/tests/integration/devices.rs:43]: Same goes for line 64, line 192 line 275, line 327line 351, line 379, line 407 and line 431

**Review by kbarkhqs — APPROVED** [review]

File: qoqo/pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "qoqo"
-version = "1.2.3"
+version = "1.2.4"
 dependencies = [
   'numpy',
   'qoqo_calculator_pyo3>=1.1',
```
[the remark this answers is not in the document]
Comment (mlodi-hqs) [reply qoqo/pyproject.toml:6]: I think we decided to stink to `x.y` versioning numbers regarding dependencies. It makes things easier to manage. But I also think that it means we should be a bit more careful with what we are doing at each point release.

**Pull request merged**